examples/fabric-admin/device_manager/CommissionerControl.h

```diff
@@ -0,0 +1,122 @@
+     * @param nodeId The node ID of the remote fabric bridge.
+     * @param endpointId The endpoint on which to send CommissionerControl commands.
+     */
+    void Init(chip::Controller::DeviceCommissioner & commissioner, chip::NodeId nodeId, chip::EndpointId endpointId);
+
+    /**
+     * @brief Sends a RequestCommissioningApproval command to the device.
+     *
+     * @param requestId The unique request ID.
+     * @param vendorId The vendor ID of the device.
+     * @param productId The product ID of the device.
+     * @param label Optional label for the device.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or an appropriate error code on failure.
+     */
+    CHIP_ERROR RequestCommissioningApproval(uint64_t requestId, uint16_t vendorId, uint16_t productId,
+                                            chip::Optional<chip::CharSpan> label);
+    /**
+     * @brief Sends a CommissionNode command to the device.
+     *
+     * @param requestId The unique request ID.
+     * @param responseTimeoutSeconds Timeout for the response in seconds.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or an appropriate error code on failure.
+     */
+    CHIP_ERROR CommissionNode(uint64_t requestId, uint16_t responseTimeoutSeconds);
+
+    /////////// CommandSender Callback Interface /////////
+    virtual void OnResponse(chip::app::CommandSender * client, const chip::app::ConcreteCommandPath & path,
+                            const chip::app::StatusIB & status, chip::TLV::TLVReader * data) override;
+
+    virtual void OnError(const chip::app::CommandSender * client, CHIP_ERROR error) override;
+
+    virtual void OnDone(chip::app::CommandSender * client) override;
+
+private:
+    enum class CommandType : uint8_t
+    {
+        kUndefined                    = 0,
+        kRequestCommissioningApproval = 1,
+        kCommissionNode               = 2,
+    };
+
+    template <class T>
+    CHIP_ERROR SendCommand(chip::DeviceProxy * device, chip::EndpointId endpointId, chip::ClusterId clusterId,
+                           chip::CommandId commandId, const T & value)
+    {
+        chip::app::CommandPathParams commandPath = { endpointId, clusterId, commandId,
+                                                     (chip::app::CommandPathFlags::kEndpointIdValid) };
+        mCommandSender = std::make_unique<chip::app::CommandSender>(this, device->GetExchangeManager(), false, false,
+                                                                    device->GetSecureSession().Value()->AllowsLargePayload());
+
+        VerifyOrReturnError(mCommandSender != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        chip::app::CommandSender::AddRequestDataParameters addRequestDataParams(chip::NullOptional);
+        // Using TestOnly AddRequestData to allow for an intentionally malformed request for server validation testing.
+        ReturnErrorOnFailure(mCommandSender->TestOnlyAddRequestDataNoTimedCheck(commandPath, value, addRequestDataParams));
```

Comment:
What is malformed about addRequestDataParams. You should NOT be calling `TestOnlyAddRequestDataNoTimedCheck`

Reply:
acked